### DISCUSSION TITLE
Relationship to Cloud Native Buildpacks?

### DISCUSSION THREAD

**Discussion by jkutner** [containerbase/base#470] [General]
Does this project have any relationship to the CNCF project https://buildpacks.io/

Is this repo an implementation of CNB, or a does it provide a component that can be used with CNBs?

**Comment by rarkins** [comment]
This project was created independently and we haven't looked into whether work could be shared in either direction

**Comment by Chumper** [comment]
It seems to me that both projects have a different vision.
https://buildpacks.io/ is about projects that will be converted to docker images based upon a given contract, 
while this project is more of a docker image where different tools can be installed in multiple versions during runtime.

**Comment by jkutner** [comment]
@Chumper @rarkins thanks for the info. I'd kindly request that you change or clarify the name. We're received questions about this. In addition, if you could put an explanation in the readme we would appreciate that.

**Comment by rarkins** [comment]
If you get questions in future maybe you can explain that the term buildpack predated either project by a long way?

**Comment by jkutner** [comment]
@rarkins The Cloud Native Buildpacks project is the third version of a technology that has been around for about 10 years: https://blog.heroku.com/buildpacks-go-cloud-native. "Buildpack" has been used by Heroku, Salesforce, Cloud Foundry, and Pivotal/VMware for about that long. Since 2018 it's been used by Google, Red Hat and others.

If this project is not a buildpack (in the sense that it is commonly used across the industry), we would appreciate you clarifying the name, and explaining this in the readme. 

Thank you for understanding. We only want to avoid confusion for our end-users.

**Reply by jkutner** [reply]
Here are some other links that explain the history and distinction:
* https://cloud.google.com/migrate/kf/docs/2.7/concepts/v2-vs-v3-buildpacks
* https://jkutner.github.io/2020/05/26/cloud-native-buildpacks-shim.html
* https://cncf-branding.netlify.app/projects/buildpacks/

**Reply by rarkins** [reply]
Thanks for the info. My impression had been that the term was pretty fragmented, however if there's a CNCF effort to unify it and Heroku is one of the companies behind it then that sounds like we should try to drop the name from our own use. Little tricky naturally as it's embedded in a bunch of downstream build systems, but we can maybe switch to the "containerbase" term as the focus.

**Reply by viceice** [reply]
@rarkins we can rename this repo/image to `base`. WDYT?

effort isn't so big

**Reply by rarkins** [reply]
Yeah, that idea came to mind too. We just need to find all the downstream uses.

**Reply by viceice** [reply]
maybe we can use our renovate package replace for them?

**Reply by jkutner** [reply]
@rarkins @viceice Have you had a chance to discuss this some more? I was reminded that the [Linux Foundation](https://www.linuxfoundation.org/) has a trademark on `buildpacks.io`, and wanted to share this information:

https://trademarks.justia.com/905/41/buildpacks-90541508.html

**Reply by viceice** [reply]
will try to do the renaming in the next two weeks

**Reply by rarkins** [reply]
That's great to hear that the term is `buildpacks.io` instead of `buildpack` in general. Thanks for your flexibility on the matter. We had been planning to rename but you've saved us a bunch of work.

**Reply by jkutner** [reply]
Sorry, I think you misunderstood me. I'm still requesting that you change the name of this project.

**Reply by viceice** [reply]
Added #524 to track, but hasn't high priority
